### ISSUE TITLE
[luxtronikheatpump] temperatureHotWaterTarget in some cases wrong

### ISSUE THREAD

**Issue by dasSofa** [openhab/openhab-addons#17132]
Alpha Innotec WZSV 62K3M Version V3.89.5 | Heatpumpt Type : MSW2-6S

The Value temperatureHotWaterTarget is in some cases not right. If i set the temperature over Openhab or the Heatpumpt itself i get the right Value for it.

I use the Smart Grid function in the Heatpump to raise my temperatureHotWaterTarget to 5°C more when there is enough Power from my PV availible.
In the case Smart Grid is active, the temperatureHotWaterTarget shows the normal value where it is set without Smart Grid (in my case 50°C ) but it should be showing 55 °C.

I have a self written PHP Script to read the Heapump Values and 2 Scripts in Node-Red (one connects over the Java Webinterface and on that uses Websockets).
All 3 of them shows the right value for temperatureHotWaterTarget with or without Smart Grid. So there is clearly a Problem in the OpenHab Luxtronik Addon here. Saddly i dont understand the Code in OpenHab so i cant help to find the Problem.

**Comment by lsiepel** [comment]
To get some details clear:

1. If you set the `temperatureHotWaterTarget` from openHAB to 50 °C, the device is adjusted all is ok.
2. The device has an internal function to alter that temp based on some conditions. If they are met:
    1. Other tools using the webinterface/websockets show 55 °C
    2. openHAB keeps showing 50 °C

It might be a bug, it might also be by design. 
There are many channels, did you check if one of those show the expected 55  °C ? 
If so this might be seperated by design to prevent a loop / unexpected raise on raise. 

Full list of channels:
https://www.openhab.org/addons/bindings/luxtronikheatpump/

**Comment by dasSofa** [comment]
No other Channel in OpenHab shows the correct Value (55 °C) when Smart Grid is active. I miss to write this in my first post, sorry!
In my OpenHab analysis / tracking all channels i couldn`t find a matching channel with the missing 5°C in my case.

But in the time of the Channel tracking i found some new Channel descriptions for OpenHab which i posted here [https://github.com/openhab/openhab-addons/issues/17043](url)

If i read out the Heatpump over the Java/Webinterface with PHP or Node-Red Channel Nr 18 is temperatureHotWaterTarget (in OpenHab) and it`s the same Value as shown in the Heapumps own Webinterface.
Only OpenHab shows 50°C when Smart Grid is active, the others show the correct Value in my case 55 °C.

**Comment by lsiepel** [comment]
Weird, there is nothing that seems to be in the way of this value being updated.
Could you check this jar (4.2.0 +) https://1drv.ms/u/s!AnMcxmvEeupwjvstXKB_A9-B1DO1Ag?e=Td4uET
It has aditional logging in place. Could you provide a trace level log  to see what values are being read after the + 5 °C should be set?

**Comment by dasSofa** [comment]
I never used the OpenHab Consol and dont understand openhab in the core or deeper level. I need to read into/learn that stuff which takes a bit of time. But i will do that.

**Comment by lsiepel** [comment]
All can be done from the UI.

1. Upgrade to openHAB 4.2.0 if not done already
2. Uninstall the current luxtronik binding
3. Drop the jar file into your addons folder
4. Enable trace logging from the binding settings in main ui.
5. add the openHAB.log to this post

**Comment by dasSofa** [comment]
Do i loose the Luxtronik items by uninstalling the binding and installing the new binding from the jar file ? In that case i dont do that because of serveral rules and the work it needs to rewrite them.

If not, i will install the binding from the jar file. BUT Smart Grid is only active/triggered from my Photovoltaik System and can not be triggered manually (For that i need to open up my Heatpump or my PV inverter and bridge that - no thanks). So we need to wait for tomorrow when my PV triggers the SmartGrid to see the 55°C.

**Comment by lsiepel** [comment]
> Do i loose the Luxtronik items by uninstalling the binding and installing the new binding from the jar file ? In that case i dont do that because of serveral rules and the work it needs to rewrite them.
> 
> If not, i will install the binding from the jar file. BUT Smart Grid is only active/triggered from my Photovoltaik System and can not be triggered manually (For that i need to open up my Heatpump or my PV inverter and bridge that - no thanks). So we need to wait for tomorrow when my PV triggers the SmartGrid to see the 55°C.

No, all Things, Items, Rules etc should just stay there. It is perfectly fine to wait a day or multiple. Just take your time. 
As always my advice is to make a backup before you do anything at all. I can;t think of anything go wrong, but it is better safe then sorry when you work on your system.

**Comment by lsiepel** [comment]
> Hi @lsiepel , i had a bad accident shortly after our conversation here. I dont know when i am able to try out your jar file and carry on with this issue.

Ai, hope you are getting better. Take your time. There are no other reports (yet), might be a specfific or isolated issue.

**Comment by lsiepel** [comment]
I hope you are fully recovered. Please let me know if there are test results.
